### PR TITLE
Downloads improvement

### PR DESCRIPTION
**Pull request by TcT2k**
A few improvements for the downloads page.

First the downloads page releases are now generated from data files which should make updating to a new version much easier. The JSON data files (`_data/release_assets/`) will be downloaded with the `update_release_asset_info` script. `_data/releases.yml` defines which versions are shown.
Download URLs, file sizes etc, are automatically generated at build time.

Additionally I've integrated the wxMSW DLL Downloads which should make getting started with them much easier, the downloads are grouped by compiler.

I've attached screenshots on how it looks now:

image

The button will open the this modal:
image

### CODE REVIEW

File: downloads/index.md [hunk 1]
```diff
@@ -1,6 +1,53 @@
 ---
 title: "Downloads"
 mirror: "https://github.com/wxWidgets/wxWidgets/releases/download"
+source_archives:
+  - description: "Windows ZIP"
+    prefix: "wxWidgets-"
+    postfix: ".zip"
+  - description: "Windows 7Z"
```
Comment (vadz) [inline downloads/index.md]: This is _really_ nitpicking, but I think `7z` is much more usual, i.e. "z" shouldn't be capitalized.

File: downloads/index.md [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+    prefix: "wxWidgets-"
+    postfix: ".7z"
+  - description: "Windows Installer"
+    prefix: "wxMSW-"
+    postfix: "-Setup.exe"
+  - description: "Source for Linux, OS X, etc"
```
Comment (vadz) [inline downloads/index.md]: Mmm, shouldn't be this be "macOS" as per the previous PR?

File: downloads/index.md [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+    postfix: ".tar.bz2"
+doc_downloads:
+  - description: "Manual (HTML) ZIP"
+    postfix: "-docs-html.zip"
+  - description: "Manual (HTML) BZIP"
+    postfix: "-docs-html.tar.bz2"
+  - description: "Manual (CHM)"
+    postfix: "-docs-chm.zip"
+compilers:
+  - description: "Visual Studio 2008"
```
Comment (vadz) [inline downloads/index.md]: Perhaps it'd better to use reverse chronological order here so that the most recent compiler is at the top?

**Comment by TcT2k** [comment]
@vadz I've updated this to include your comments and updated changed the update script to ruby which makes the update process even easier.
I'll create a PR with updated instructions in the main repo

**Comment by tierra** [comment]
Looks good to me, great idea pulling release info from GitHub API.

**Comment by vadz** [comment]
Thanks again Tobias!

**Pull request merged**